src/jit/lsra.h

```diff
@@ -538,6 +538,22 @@ class LinearScan : public LinearScanInterface
     LsraSpill                   getLsraSpill()                  { return (LsraSpill) (lsraStressMask & LSRA_SPILL_MASK); }
     bool                        spillAlways()                   { return getLsraSpill() == LSRA_SPILL_ALWAYS; }
 
+    // This controls whether RefPositions that require a register optionally should
+    // be allocated a busy register by spilling another less important RefPosition.
+    enum LsraRegOptionalControl     { LSRA_REG_OPTIONAL_DEFAULT          = 0,
+                                      LSRA_REG_OPTIONAL_NO_BUSYREG_ALLOC = 0x1000,
+                                      LSRA_REG_OPTIONAL_MASK             = 0x1000 };
+
+    LsraRegOptionalControl      getLsraRegOptionalControl()            
+    { 
+        return (LsraRegOptionalControl) (lsraStressMask & LSRA_REG_OPTIONAL_MASK); 
+    }
+
+    bool                        regOptionalNoBusyRegAlloc()     
+    { 
+        return getLsraRegOptionalControl() == LSRA_REG_OPTIONAL_NO_BUSYREG_ALLOC;  
+    }
+
```

Comment:
Ok. I have renamed it as LSRA_REG_OPTIONAL_NO_ALLOC and added logic in allocateRegisters() not to attempt to allocate a reg to reg-optional ref positions under stress mode.